llvm/lib/Analysis/CmpInstAnalysis.cpp

```diff
@@ -96,10 +97,27 @@ llvm::decomposeBitTestICmp(Value *LHS, Value *RHS, CmpInst::Predicate Pred,
     Pred = ICmpInst::getStrictPredicate(Pred);
   }
 
+  auto decomposeBitMask =
```

Comment:
I am confused why the original code doesn't handle this pattern.
cc @andjo403

Reply:
Answered in https://github.com/llvm/llvm-project/pull/136367

"Historically the purpose of decomposeBitTestICmp() was to decompose plain icmps into and+icmp form, but I don't see a reason not to support the and+icmp form directly for cases where the distinction does not matter."